sdk/storage/storage-file-share/src/ShareFileClient.ts

```diff
@@ -1400,31 +1400,37 @@ export class ShareFileClient extends StorageClient {
   // High Level functions
 
   /**
-   * ONLY AVAILABLE IN BROWSERS.
-   *
-   * Uploads a browser Blob/File/ArrayBuffer/ArrayBufferView object to an Azure File.
+   * Uploads a Buffer(Node)/Blob/ArrayBuffer/ArrayBufferView to an Azure File.
    *
-   * @param {Blob | ArrayBuffer | ArrayBufferView} browserData Blob, File, ArrayBuffer or ArrayBufferView
+   * @param {Buffer | Blob | ArrayBuffer | ArrayBufferView} data Buffer(Node), Blob, ArrayBuffer or ArrayBufferView
    * @param {FileParallelUploadOptions} [options]
    * @returns {Promise<void>}
    */
-  public async uploadBrowserData(
-    browserData: Blob | ArrayBuffer | ArrayBufferView,
+  public async uploadData(
+    data: Buffer | Blob | ArrayBuffer | ArrayBufferView,
     options: FileParallelUploadOptions = {}
   ): Promise<void> {
-    const { span, spanOptions } = createSpan(
-      "ShareFileClient-uploadBrowserData",
-      options.tracingOptions
-    );
+    const { span, spanOptions } = createSpan("ShareFileClient-uploadData", options.tracingOptions);
     try {
-      const browserBlob = new Blob([browserData]);
-      return await this.uploadSeekableBlob(
-        (offset: number, size: number): Blob => {
-          return browserBlob.slice(offset, offset + size);
-        },
-        browserBlob.size,
-        { ...options, tracingOptions: { ...options!.tracingOptions, spanOptions } }
-      );
+      if (isNode && data instanceof Buffer) {
+        return this.uploadBuffer(
+          (offset, count) => Buffer.from(data.subarray(offset, offset + count)),
+          data.byteLength,
+          {
+            ...options,
```

Comment:
New version of uploadData method (to avoid extra copy operation of buffer by "stream-buffers")-
- Instead of calling the `uploadResetableStream`, it now calls `uploadBuffer` which is similar to `uploadResetableStream` except for passing in a [bufferChunk](https://github.com/Azure/azure-sdk-for-js/pull/6060/files#r344023521) without the usage of ReadStream.